packages/salesforcedx-vscode-core/test/jest/commands/templates/forceApexUnitClassCreate.test.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { forceApexUnitClassCreate } from '../../../../src/commands/templates';
+import { LibraryForceApexUnitClassCreateExecutor } from '../../../../src/commands/templates/executors/LibraryForceApexUnitClassCreateExecutor';
+import {
+  APEX_CLASS_DIRECTORY,
+  APEX_CLASS_NAME_MAX_LENGTH,
+  APEX_CLASS_TYPE
+} from '../../../../src/commands/templates/metadataTypeConstants';
+import { OverwriteComponentPrompt } from '../../../../src/commands/util/overwriteComponentPrompt';
+import {
+  CompositeParametersGatherer,
+  MetadataTypeGatherer,
+  SelectFileName,
+  SelectOutputDir
+} from '../../../../src/commands/util/parameterGatherers';
+import * as commandlet from '../../../../src/commands/util/sfdxCommandlet';
+import { SfdxWorkspaceChecker } from '../../../../src/commands/util/sfdxWorkspaceChecker';
+
+jest.mock('../../../../src/commands/templates/executors/LibraryForceApexUnitClassCreateExecutor');
+jest.mock('../../../../src/commands/util/overwriteComponentPrompt');
+jest.mock('../../../../src/commands/util/parameterGatherers');
+jest.mock('../../../../src/commands/util/sfdxWorkspaceChecker');
+jest.mock('../../../../src/commands/util/timestampConflictChecker');
+
+const selectFileNameMocked = jest.mocked(SelectFileName);
+const metadataTypeGathererMocked = jest.mocked(MetadataTypeGatherer);
+const selectOutputDirMocked = jest.mocked(SelectOutputDir);
+const libraryForceApexUnitClassCreateExecutorMocked = jest.mocked(
+  LibraryForceApexUnitClassCreateExecutor
+);
+const sfdxWorkspaceCheckerMocked = jest.mocked(SfdxWorkspaceChecker);
+const compositeParametersGathererMocked = jest.mocked(
+  CompositeParametersGatherer
+);
+const overwriteComponentPromptMocked = jest.mocked(OverwriteComponentPrompt);
+
```

Comment:
One interesting thing about this test. It was failing for me for a while and blowing my mind b/c I was pretty much following the pattern found in the create apex class test.  Turned out it came down to a problem with using core module imports vs salesforcedx-vscode-utils imports in the executor.   Nothing really to gain from this comment other then the reminder we should fix that problem have having duplicates.